Extensions/Multiplayer/multiplayertools.ts

```diff
@@ -1519,6 +1626,125 @@ namespace gdjs {
       );
     };
 
+    const onLobbyQuickJoinFinished = (runtimeScene: gdjs.RuntimeScene) => {
+      _isJoiningOrStartingAGame = false;
+      gdjs.multiplayerComponents.displayLoader(runtimeScene, false);
+    };
+
+    const doQuickJoinLobby = async (
+      runtimeScene: gdjs.RuntimeScene,
+      displayLoader: boolean
+    ) => {
+      if (_isJoiningOrStartingAGame) return;
+
+      const _gameId = gdjs.projectData.properties.projectUuid;
+      if (!_gameId) {
+        handleLobbiesError(
+          runtimeScene,
+          'The game ID is missing, the quick join lobby action cannot continue.'
+        );
+        return;
+      }
+
+      _isJoiningOrStartingAGame = true;
+      if (displayLoader) {
+        gdjs.multiplayerComponents.displayLoader(runtimeScene, true);
+      }
+
+      const quickJoinLobbyRelativeUrl = `/play/game/${_gameId}/public-lobby/action/quick-join`;
+
+      try {
+        const quickJoinLobbyResponse: QuickJoinLobbyResponse = await gdjs.evtTools.network.retryIfFailed(
+          { times: 2 },
+          () =>
+            // TODO: should the supported compression methods be sent to the endpoint?
```

Comment:
I think so. It seems only fair to suggest a lobby to connect to, only if it's compatible.

Reply:
Done